File: src/Element.java
import java.awt.Color;

/**
 * Remplacement de l'ancienne classe Element par un enum.
 * Chaque constante porte sa couleur et son effet appliqué sur un monstre touché.
 * getCouleur() est maintenu pour compatibilité avec Tours.drawVisuel().
 */
public enum Element {

    FEU {
        @Override public Color getCouleur() { return new Color(184, 22, 1); }
        @Override public void appliquerEffet(Monstres cible) {
            cible.burnDamagePerSec  = 2.0;
            cible.burnTimeRemaining = 3.0;
        }
    },
    EAU {
        @Override public Color getCouleur() { return new Color(6, 0, 160); }
        @Override public void appliquerEffet(Monstres cible) {
            cible.slowFactor       = 0.5;
            cible.slowTimeRemaining = 2.0;
        }
    },
    TERRE {
        @Override public Color getCouleur() { return new Color(0, 167, 15); }
        @Override public void appliquerEffet(Monstres cible) {
            cible.armorReduction      = 1.25;
            cible.armorTimeRemaining  = 3.0;
        }
    },
    VENT {
        @Override public Color getCouleur() { return new Color(173, 216, 230); }
        @Override public void appliquerEffet(Monstres cible) {
            cible.recoilCases++;
        }
    },
    NEUTRE {
        @Override public Color getCouleur() { return Color.GRAY; }
        @Override public void appliquerEffet(Monstres cible) {}
    };

    public abstract Color getCouleur();
    public abstract void appliquerEffet(Monstres cible);
}

File: activity/client/src/game/Element.ts
import type { Monstre } from "./Monstres";

export enum Element { FEU = "FEU", EAU = "EAU", TERRE = "TERRE", VENT = "VENT", NEUTRE = "NEUTRE" }

export function getCouleur(e: Element): string {
  switch (e) {
    case Element.FEU:   return "#B81601";
    case Element.EAU:   return "#0600A0";
    case Element.TERRE: return "#00A70F";
    case Element.VENT:  return "#ADD8E6";
    default:            return "#888888";
  }
}

export function appliquerEffet(e: Element, m: Monstre): void {
  switch (e) {
    case Element.FEU:
      m.burnDamagePerSec  = 2.0;
      m.burnTimeRemaining = 3.0;
      break;
    case Element.EAU:
      m.slowFactor        = 0.5;
      m.slowTimeRemaining = 2.0;
      break;
    case Element.TERRE:
      m.armorReduction    = 1.25;
      m.armorTimeRemaining= 3.0;
      break;
    case Element.VENT:
      m.recoilCases++;
      break;
  }
}
